### DISCUSSION TITLE
expanding image with last row of pixels

### DISCUSSION THREAD

**Discussion by stiebel** [ImageMagick/ImageMagick#3483] [Help]
Hi, I need some help with large images that shall be printed and cut.
These images are - lets say - 3000 pixels wide an will be printed with 300dpi resulting 10 inch width.
the paper is 11 inch wide and the image is placed in the middle.
This images shall be cut automatically and the "knife" is very precise, but not perfect. 
So I want to expand the image without scaling so that the first row of pixels shall be repeated lets say 20 times
and stitched at the beginning and the last row of pixels shall be repeated 20 times and stitched at the end of the image
resulting in an image with a width of 3040 pixels.
So if the "knife" moves for 1 or 2 pixels to the side, I will never see white paper and the image is still
unscaled and with exact aspect ratio.
Is there an "expand" command for this? 
If not, is there a good way of doing it with a script?
Thanks for your help!
Roland

**Comment by snibgo** [comment]
A "null" distort with a viewport can do that. For example, expanding 10 pixels to the right and 20 to the bottom:
```
magick toes.png -set option:distort:viewport %[fx:w+10]x%[fx:h+20]+0+0 -distort SRT 1,0 +repage x.png
```
![x](https://user-images.githubusercontent.com/33812368/113147644-ded85a00-9228-11eb-8eee-36a3a45c8465.png)

EDIT to add:

If you want extra pixels at left as well as right, and top as well as bottom, then use negative offsets. Foe example:
```
magick toes.png -set option:distort:viewport %[fx:w+10]x%[fx:h+20]-2-3 -distort SRT 1,0 +repage x.png
```
This increases width by 10 and height by 20, with 2 of the extra width pixels on the left, and 3 of the extra height pixels at the top.

EDIT 2: I suggest you have "+repage" after any distort.

**Comment by stiebel** [comment]
Hi snigbo,
this is exactly what I was looking for.
(by the way, I use "convert" instead of "magick" as "magick" is not recognized by my system...)
Thanks a lot!

ImageMagick is great!
Roland
